weather_dl/download_pipeline/parsers.py

```diff
@@ -326,6 +336,36 @@ def _number_of_replacements(s: t.Text):
     return len(set(format_names)) + num_empty_names
 
 
+def _validate_target_path(config: Config):
+    """
+    Checks accessibility of target locations before execution start
+    """
+    client = storage.Client()
+    for partition_conf in prepare_partitions(config):
+        target = prepare_target_name(partition_conf)
+        parsed = urlparse(target)
+        if parsed.scheme == 'gs':
```

Comment:
Is there a way this this validation could be written to make the check portable (to other clouds)? For example, via beam's FileSystem's API?

Reply:
That's a great point. I figured the "path writeable" check needs to distinguish between file systems so stuck to GCS. Nevertheless `FileSystem` does have an FS-agnostic way for checking whether the directory exists. I've updated the CL to support this.